homeassistant/components/aemet/abstract_aemet_sensor.py

```diff
@@ -0,0 +1,77 @@
+"""Abstraction form AEMET OpenData sensors."""
+from homeassistant.const import ATTR_ATTRIBUTION
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+
+from .const import ATTRIBUTION, SENSOR_DEVICE_CLASS, SENSOR_NAME, SENSOR_UNIT
+
+
+class AbstractAemetSensor(Entity):
+    """Abstract class for an AEMET OpenData sensor."""
+
+    def __init__(
+        self,
+        name,
+        unique_id,
+        sensor_type,
+        sensor_configuration,
+        coordinator: DataUpdateCoordinator,
```

Comment:
You might as well make this a weather_coordinator since you have your own baseclass.

Reply:
This is also based on OpenWeatherMap, but I'll look into it:
https://github.com/home-assistant/core/blob/20f45f8ab9e744de7488fe66c5e7f8aca9b60e2d/homeassistant/components/openweathermap/abstract_owm_sensor.py#L18